### ISSUE TITLE
Add artifact.push default to global config

### ISSUE THREAD

**Issue by r2d4** [GoogleContainerTools/skaffold#1172]
Currently the push/don't push default value is set by the logic

```
if context == minikube || context == docker-for-desktop 
then don't push by default

else push by default
```

This makes sense to be configurable on a per context basis.

Would address https://github.com/GoogleContainerTools/skaffold/issues/1158 and other local clusters that aren't magically named minikube or docker-for-desktop.

**Comment by corneliusweig** [comment]
@r2d4 #1355 keeps the special cases for known local contexts, but makes the artifact.push behavior configurable in the global config. Does that resolve this issue?

**Comment by nkubala** [comment]
Fixed by #1355
